ls1intum/Artemis
# `General`: Add a plagiarism management overview including notifications

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


This allows instructors to notify students to about potential plagiarism cases. The students can give statements to the plagiarism case and the instructor can decide whether to deny or confirm them.

### Description

I took this pull request over from another developer, so I can't really tell what the changes are in detail but there's a lot of them. 😵‍💫 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Text oder modeling exercise with a detected plagiarism

1a.Log in to Artemis as Instructor
1b. Navigate to Course Detail
1c. Open Plagiarism Cases
1d. Asses the plagiarism case and send notification

2a. Log in to Artemis as Student
2b. Open Notifications
2c. Click on the Plagiarism Notification
2d. Respond to the plagiarism Case

3a. Login in again to Artemis as Instructor
3b. Navigate to Course Detail
3c. Open Plagiarism Cases
3d. Assess the student's response

4a. Log in to Artemis again as Student
4b. Open Notifications
4c. Click on the Plagiarism Notification
4d. Check that the status has been updated by the instructor


### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots


<img width="1431" alt="Bildschirmfoto 2021-11-14 um 12 39 39" src="https://user-images.githubusercontent.com/51987021/141679274-5663983d-7153-4971-9e8b-4031193a1e8d.png">
<img width="1431" alt="Bildschirmfoto 2021-11-14 um 12 39 08" src="https://user-images.githubusercontent.com/51987021/141679276-1ab85fe0-8d02-419c-bbd5-253d9f32edd8.png">
<img width="261" alt="Bildschirmfoto 2021-11-14 um 12 38 52" src="https://user-images.githubusercontent.com/51987021/141679277-e0a35852-b7dd-49dc-ba36-3d555506982a.png">
<img width="301" alt="Bildschirmfoto 2021-11-14 um 12 38 38" src="https://user-images.githubusercontent.com/51987021/141679278-1326b08a-d34e-4f47-bdcd-09c3682ccb5f.png">
